packages/votingV2/src/mappings/votingV2.ts

```diff
@@ -117,42 +136,127 @@ export function handlePriceResolved(event: PriceResolved): void {
   requestRound.roundId = event.params.roundId;
   requestRound.votersEligibleForRewardsRatio = voterGroup.votersAmount.div(requestRound.votersAmount);
 
-  requestRound.votersEligibleForRewardsPercentage =
-    requestRound.votersEligibleForRewardsRatio.times(BIGDECIMAL_HUNDRED);
+  requestRound.votersEligibleForRewardsPercentage = defaultBigDecimal(requestRound.votersEligibleForRewardsRatio).times(
+    BIGDECIMAL_HUNDRED
+  );
   requestRound.winnerGroup = voterGroup.id;
   requestRound.gat = roundInfo.reverted ? requestRound.gat : toDecimal(roundInfo.value.value0);
 
-  requestRound.gatPercentageRaw = requestRound.gat.div(toDecimal(getTokenContract().try_totalSupply().value));
-  requestRound.gatPercentage = requestRound.gatPercentageRaw.times(BIGDECIMAL_HUNDRED);
-  requestRound.cumulativeActiveStakeAtRound = cumulativeActiveStakeAtRound;
+  requestRound.gatPercentageRaw = defaultBigDecimal(requestRound.gat).div(
+    toDecimal(getTokenContract().try_totalSupply().value)
+  );
+  requestRound.gatPercentage = defaultBigDecimal(requestRound.gatPercentageRaw).times(BIGDECIMAL_HUNDRED);
+  requestRound.cumulativeStakeAtRound = cumulativeStakeAtRound;
+
+  let globals = getOrCreateGlobals();
+  let users = globals.userAddresses;
 
   requestRound.save();
   request.save();
   voterGroup.save();
+
+  let slashingTrackers = votingContract.try_requestSlashingTrackers(request.requestIndex);
+
+  for (let i = 0; i < users.length; i++) {
```

Comment:
this is a LONG function now and does a fair bit of complex stuff! can I ask two things:
1) let's add a few more comments. this looks like it might be a little bit difficult to maintain otherwise.
2) prehaps we use more `log.x` to construct more debug logs

Reply:
Thank you, I have decided to split the function to improve readability and added comments.